src/support/utils.h

```diff
@@ -29,75 +29,31 @@ int cmp_lt(void *a, numerictype_t atag, void *b, numerictype_t btag);
 int cmp_eq(void *a, numerictype_t atag, void *b, numerictype_t btag,
            int equalnans);
 
-#ifdef __x86_64__
-#  define LEGACY_REGS "=Q"
+#if defined(__clang__) || (defined(__GNUC__) && (__GNUC__ > 4 || __GNUC_MINOR__ >= 8))
+#define bswap_16(x) __builtin_bswap16(x)
```

Comment:
This seems to be available in https://gcc.gnu.org/onlinedocs/gcc-4.5.0/gcc/Other-Builtins.html which is older than our minimum gcc version requirement so remove the version check (or at least use it on 4.7)?